dropwizard-validation/src/test/java/io/dropwizard/validation/ConstraintViolationsTest.java

```diff
@@ -0,0 +1,108 @@
+package io.dropwizard.validation;
+
+import io.dropwizard.validation.valuehandling.OptionalValidatedValueUnwrapper;
+import org.hibernate.validator.HibernateValidator;
+import org.junit.Test;
+
+import javax.validation.*;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.executable.ExecutableValidator;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SuppressWarnings({"UnusedDeclaration"})
+public class ConstraintViolationsTest {
+
+    private final Validator validator = Validation
+            .byProvider(HibernateValidator.class)
+            .configure()
+            .addValidatedValueHandler(new OptionalValidatedValueUnwrapper())
+            .buildValidatorFactory()
+            .getValidator();
+
+    private final ExecutableValidator execValidator = validator.forExecutables();
+
+    @Max(3)
+    public Integer theIntTest(@Min(2) Integer x) {
+        return x + 1;
+    }
+
+    @Valid
+    public ConstraintPerson thePersonTest(@Valid ConstraintPerson person) {
+        person.setName(null);
+        return person;
+    }
+
+    @Test
+    public void invalidIntReturnIs500() throws NoSuchMethodException {
+        Set<ConstraintViolation<ConstraintViolationsTest>> violations =
+                execValidator.validateReturnValue(
+                        new ConstraintViolationsTest(),
+                        ConstraintViolationsTest.class.getMethod("theIntTest", Integer.class),
+                        4 // the return value
+                );
+
+        assertThat(violations).hasSize(1);
+
+        Set<ConstraintViolation<?>> copy = ConstraintViolations.copyOf(violations);
+        assertThat(ConstraintViolations.determineStatus(copy)).isEqualTo(500);
```

Comment:
Would that work? A `ConstraintViolationException` `.getConstraintViolations()` returns a `Set<ConstraintViolation<?>>`, and that would make `determineStatus` too narrow  to use. Please advise.

Implemented your other suggestions.